docs/design/features/lsra-detail.md

```diff
@@ -393,109 +384,110 @@ the `LsraBlockInfo`.
+constructed as described above. This traversal invokes `LinearScan::BuildNode()` for each
+node, which builds `RefPositions` according the the liveness model described above:
 
-    -   This method is responsible for initializing the `NodeInfo` for
-        each node.
+-   First, we create `RefPosition`s to define any internal registers that are required.
+    As we create new `Interval`s for these, we add the definition `RefPosition` to an array,
+    `InternalDefs`. This allows us to create the corresponding uses of these internal
+    registers later.
 
-        -   This is currently maintained in the `gtLsraInfo` for each
-            node.
+-   Then we create `RefPosition`s for each use in the instruction.
 
-        -   The nodes are placed into a map, that maps from a tree node
-            to the list of tree nodes that are the actual operands.
+    -   A use of a register candidate lclVar becomes a `RefTypeUse` `RefPosition` on the 
+        `Interval` associated with the lclVar.
 
-            -   This complication is due to contained nodes, where, e.g.
-                a `GT_IND` node may be contained, and have a
-                `GT_LEA` child whose base and addr nodes are the
-                actual operands.
+    -   For tree-temp operands (including non-register-candidate lclVars), we may have one
+        of 3 situations:
 
-            -   Work is underway to eliminate the `gtLsraInfo`, and to
-                place the `NodeInfo` of the actual operands directly
-                into the map, eliminating the need for two level (map of
-                lists) data structure [Issue \#7225](https://github.com/dotnet/coreclr/issues/7225)
+         - A contained immediate requires no registers, so no `RefPosition`s are created.
 
-            -   Subsequent work will eliminate the `NodeInfo`
-                altogether, and instead build`RefPosition`s directly
-                from the `TreeNodeInfoInit` methods [Issue \#7257](https://github.com/dotnet/coreclr/issues/7257)
+         - A contained memory operand or addressing mode will cause `RefPosition`s to be
+           created for any (non-contained) base or index registers.
+         - A single `RefPosition` is created for non-contained nodes.
 
-    -   The `NodeInfo` includes:
+         In order to build these uses, we need to find the `Interval` associated with the
+         tree-temp. This is done via the `defList`, which contains the `RefTypeDef` `RefPosition`
+         for all tree temps whose def has been encountered but for which we
+         have not yet seen the use. This is a simple list on the assumption that the distance
+         between defs and uses of tree temps is rarely very great.
 
-        -   The number of register sources and destinations.
+         For x86 and x64, when we have an instruction that will overwrite one of its sources,
+         we need to ensure that the other source isn't assigned the same register as the
```

Comment:
I think it's more correct as-is, because we think of assigning a register to a node or tree rather than vice-versa, but perhaps "isn't given the same register" would be clearer.